action.yml

```diff
@@ -95,13 +101,13 @@ runs:
 
     - name: Install project (no extras or groups)
       if: inputs.extras == '' && inputs.groups == ''
-      run: poetry install --no-interaction
+      run: poetry install --no-interaction ${{ inputs.install-project == 'true' && '' || '--no-root' }}
       shell: bash
 
     - name: Install project with --extras=${{ inputs.extras }} --with=${{ inputs.groups }}
       if: inputs.extras != '' || inputs.groups != ''
       # (Empty extras or groups lists are fine.)
-      run: poetry install --no-interaction --extras="${{ inputs.extras }}" --with="${{ inputs.groups }}"
+      run: poetry install --no-interaction --extras="${{ inputs.extras }}" --with="${{ inputs.groups }}" ${{ inputs.install-project == 'true' && '' || '--no-root' }}
       shell: bash
```

Comment:
Was a bit confused by why we have these two steps here. I think it's because I couldn't find a cleaner way to only add `--extras` if `inputs.extras` was specified: I don't think there's a ternary syntax or [helper function](https://docs.github.com/en/actions/learn-github-actions/expressions)?